### PR TITLE
Feat(eos_cli_config_gen): Add IPsec and NAT options to tunnel_interfaces.

### LINKED ISSUE #3824: Feat(eos_cli_config_gen): Add IPsec and NAT options to `tunnel_interfaces`
### Enhancement summary

Some new functionality is needed for tunnel interfaces

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Want to configure this:

```
interface Tunnel99
   mtu 1394
   ip address 20.99.1.1/24
   ip nat service-profile vrf-aware-nat  <<<<<<<<<<<<<<<<<<< missing
   tunnel mode ipsec                               <<<<<<<<<<<<<<<<<<< missing
   tunnel source interface Ethernet2
   tunnel destination 199.168.148.132
   tunnel ipsec profile ipsec00                  <<<<<<<<<<<<<<<<<<< missing
!
```

### Describe the solution you would like

Add new keys to the model and relevant template / documentation

```
tunnel_interfaces:

    # Tunnel Interface Name
  - name: <str; required; unique>
    description: <str>
    # NEW KEYS BELOW
    tunnel_mode: < [ipsec, gre] ; no default>
    ipsec_profile:  <str> # Add a description stating it is used only when tunnel_mode is set to ipsec and that it must target a defined IPsec profile
    nat_profile: <str> # Same it must target an existing NAT profile
    # END OF NEW KEYS
    # Keep eos_cli at the end
```

### Describe alternatives you have considered

`eos_cli` but need better

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Add ip nat service-profile, tunnel mode and tunnel ipsec profile to tunnel interfaces.

## Related Issue(s)

Fixes #3824 

## Component(s) name

`arista.avd.eos_cli_config_gen`

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tunnel-interfaces.j2 [hunk 1]
```diff
@@ -64,6 +70,9 @@ interface {{ tunnel_interface.name }}
 {%     if tunnel_interface.path_mtu_discovery is arista.avd.defined(true) %}
    tunnel path-mtu-discovery
 {%     endif %}
+{%     if tunnel_interface.ipsec_profile is arista.avd.defined and tunnel_interface.tunnel_mode == "ipsec" %}
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tunnel-interfaces.j2:73]: shouldn't we check that ip sec profile is there in ip_security.profiles. If profile is not there in device the eapi playbook fails.
```
fatal: [wan2-dc1]: FAILED! => {"changed": false, "module_stderr": "IPsec profile Profile-2 has not been created", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error"}
```

**Review by MaheshGSLAB — APPROVED** [review]
LGTM, tested on lab.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml [hunk 2]
```diff
@@ -76,6 +76,13 @@ keys:
               valid_values: ["ingress", "egress"]
               description: |
                 Optional direction ('ingress', 'egress')  for tcp mss ceiling
+        tunnel_mode:
+          type: str
+          valid_values: ["gre", "ipsec"]
+          description: |
+            Tunnel encapsulation method.
+            gre: Generic route encapsulation protocol,
+            ipsec: IPsec-over-IP encapsulation.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml]:
```suggestion
            `gre`: Generic route encapsulation protocol,
            `ipsec`: IPsec-over-IP encapsulation.
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml [hunk 3]
```diff
@@ -76,6 +76,13 @@ keys:
               valid_values: ["ingress", "egress"]
               description: |
                 Optional direction ('ingress', 'egress')  for tcp mss ceiling
+        tunnel_mode:
+          type: str
+          valid_values: ["gre", "ipsec"]
+          description: |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml]:
```suggestion
          description: |-
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml [hunk 4]
```diff
@@ -85,6 +92,14 @@ keys:
         path_mtu_discovery:
           type: bool
           description: Enable Path MTU Discovery On Tunnel
+        ipsec_profile:
+          type: str
+          description: |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml]:
```suggestion
          description: |-
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml [hunk 5]
```diff
@@ -85,6 +92,14 @@ keys:
...
+            Used only when tunnel_mode is set to ipsec.
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/tunnel_interfaces.schema.yml]:
```suggestion
            Used only when `tunnel_mode` is set to `ipsec`.
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tunnel-interfaces.j2 (hunk 1 above)
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/tunnel-interfaces.j2]: We should not have logic in eos_cli_config_gen. Just configure what is given as long as it is accepted by EOS.
```suggestion
{%     if tunnel_interface.ipsec_profile is arista.avd.defined %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tunnel-interfaces.j2 [hunk 6]
```diff
@@ -25,10 +25,16 @@
...
+{%         set nat_profile = tunnel_interface.nat_profile | arista.avd.default('-') %}
+{%         set tunnel_mode = tunnel_interface.tunnel_mode | arista.avd.default("-") %}
 {%         set row_source_interface = tunnel_interface.source_interface | arista.avd.default("-") %}
 {%         set row_destination = tunnel_interface.destination | arista.avd.default("-") %}
 {%         set row_pmtu = tunnel_interface.path_mtu_discovery | arista.avd.default("-") %}
-| {{ tunnel_interface.name }} | {{ description }} | {{ vrf }} | {{ mtu }} | {{ shutdown }} | {{ row_source_interface }} | {{ row_destination }} | {{ row_pmtu }} |
+{%         set ipsec_profile = "-" %}
+{%         if tunnel_mode == "ipsec" %}
+{%             set ipsec_profile = tunnel_interface.ipsec_profile | arista.avd.default("-") %}
+{%         endif %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tunnel-interfaces.j2]:
```suggestion
{%         set ipsec_profile = tunnel_interface.ipsec_profile | arista.avd.default("-") %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tunnel-interfaces.j2 [hunk 7]
```diff
@@ -25,10 +25,14 @@
 {%         set vrf = tunnel_interface.vrf | arista.avd.default('default') %}
 {%         set mtu = tunnel_interface.mtu | arista.avd.default('-') %}
 {%         set shutdown = tunnel_interface.shutdown | arista.avd.default('-') %}
+{%         set nat_profile = tunnel_interface.nat_profile | arista.avd.default('-') %}
+{%         set tunnel_mode = tunnel_interface.tunnel_mode | arista.avd.default("-") %}
 {%         set row_source_interface = tunnel_interface.source_interface | arista.avd.default("-") %}
 {%         set row_destination = tunnel_interface.destination | arista.avd.default("-") %}
 {%         set row_pmtu = tunnel_interface.path_mtu_discovery | arista.avd.default("-") %}
-| {{ tunnel_interface.name }} | {{ description }} | {{ vrf }} | {{ mtu }} | {{ shutdown }} | {{ row_source_interface }} | {{ row_destination }} | {{ row_pmtu }} |
+{%         set ipsec_profile = "-" %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/tunnel-interfaces.j2]:
```suggestion
```

**Pull request merged**